scripts/tests/run_tv_casting_test.py

```diff
@@ -80,36 +87,53 @@ def dump_temporary_logs_to_console(log_file_path: str):
 
 def handle_casting_failure(casting_state: str, log_file_paths: List[str]):
     """Log '{casting_state} failed!' as error, dump log files to console, exit on error."""
-    logging.error(casting_state + ' failed!')
+    logging.error(f'{casting_state} failed!')
 
     for log_file_path in log_file_paths:
         try:
             dump_temporary_logs_to_console(log_file_path)
         except Exception as e:
-            logging.exception(f"Failed to dump {log_file_path}: {e}")
+            logging.exception(f'Failed to dump {log_file_path}: {e}')
 
     sys.exit(1)
 
 
-def extract_value_from_string(line: str) -> str:
+def extract_value_from_string(line: str, value_name: str, casting_state: str, log_paths) -> str:
     """Extract and return value from given input string.
 
     The string is expected to be in the following format as it is received
     from the Linux tv-casting-app output:
+    \x1b[0;34m[1715206773402] [20056:2842184] [DMG]  Cluster = 0x506,\x1b[0m
+    The substring to be extracted here is '0x506'.
+    Or:
+    \x1b[0;32m[1714582264602] [77989:2286038] [SVR] Discovered Commissioner #0\x1b[0m
+    The integer value to be extracted here is '0'.
+    Or:
     \x1b[0;34m[1713741926895] [7276:9521344] [DIS] Vendor ID: 65521\x1b[0m
-    The integer value to be extracted here is 65521.
+    The integer value to be extracted here is '65521'.
     Or:
     \x1b[0;34m[1714583616179] [7029:2386956] [SVR] 	device Name: Test TV casting app\x1b[0m
     The substring to be extracted here is 'Test TV casting app'.
     """
-    value = line.split(':')[-1].strip().replace('\x1b[0m', '')
+    if ':' in line:
+        if '=' in line:
+            delimiter = '='
+        elif '#' in line:
+            delimiter = '#'
+        else:
+            delimiter = ':'
+
+        value = line.split(delimiter)[-1].strip().replace('\x1b[0m', '').rstrip(',')
+    else:
+        logging.error(f'Could not extract {value_name} from the following line: {line}')
+        handle_casting_failure(casting_state, log_paths)
```

Comment:
how about throwing an exception and letting the caller handle it? Passing in casting_state and log_paths just for failure handling seems inconvenient - many args which are not relevant to the functioning of this function.

Reply:
Good call. Addressed in the latest commit.